kernel/src/dynamic_process_loading.rs

```diff
@@ -0,0 +1,1124 @@
+                                for (proc_index, value) in procs.iter().enumerate() {
+                                    if proc_index < new_process_count {
+                                        {
+                                            // check if there is a remnant app in that space
+                                            if new_address
+                                                == value.unwrap().get_addresses().flash_start
+                                            {
+                                                // indicates there is an active process whose binary is stored here
+                                                is_remnant_region = false;
+                                                // Because there is an app which is also an active process, we move to the next address
+                                                new_address += cmp::max(app_length, size);
+                                            }
+                                        }
+                                    }
+                                }
+                            });
+                        }
+                    }
+                    Err(_e) => {
+                        return Err(ErrorCode::FAIL);
+                    }
+                }
+            }
+
+            if is_padding_app || is_empty_region || is_remnant_region {
+                let address_validity_check = self.check_overlap_region(test_address, app_length);
+
+                match address_validity_check {
+                    Ok(()) => {
+                        // despite doing all these, if the new app's start address and size make it such that it will
+                        // cross the bounds of flash, we return a No Memory error. (this is currently untested)
+                        if new_address + (app_length - 1) > flash_end {
+                            return Err(ErrorCode::NOMEM);
+                        }
+                        // otherwise, we found the perfect address for our new app
+                        return Ok(new_address);
+                    }
+                    Err((new_start_addr, _e)) => {
+                        // We try again from the end of the overlapping app
+                        new_address = new_start_addr;
+                    }
+                }
+            }
+        }
+        Err(ErrorCode::NOMEM)
+    }
+
+    //********************* Loading Process into Process Array **************************//
+
+    fn load_processes(
+        &self,
+        app_flash: &'static [u8],
+        app_memory: &'static mut [u8],
+        _capability_management: &dyn ProcessManagementCapability,
+    ) -> Result<(), ProcessLoadError> {
+        let (remaining_memory, _remaining_flash) =
+            self.load_processes_from_flash(app_flash, app_memory)?;
+
+        if config::CONFIG.debug_process_credentials {
+            debug!("Checking: no checking, load and run all processes");
```

Comment:
I don't follow, how do we know there is no checking?

Reply:
I have not implemented credential checking yet.

Reply:
Can we not just use the existing process loader, which should do the checking for us? It seems strange to duplicate the loading

Reply:
No, the current synchronous process loader does not perform process checking anyway. Plus we lose access to the current processes array after loading the new app if we do not duplicate the functions.

Reply:
Why can't we just modify the existing ones to not loose access?